src/Storages/MergeTree/DataPartsExchange.cpp

```diff
@@ -580,9 +580,8 @@ MergeTreeData::MutableDataPartPtr Fetcher::downloadPartToMemory(
         new_projection_part->is_temp = false;
         new_projection_part->setColumns(block.getNamesAndTypesList());
         MergeTreePartition partition{};
-        IMergeTreeDataPart::MinMaxIndex minmax_idx{};
         new_projection_part->partition = std::move(partition);
-        new_projection_part->minmax_idx = std::move(minmax_idx);
```

Comment:
It is not clear why do we use `shared_ptr` for `minmax_idx` now.
Also maybe unique can do?

Reply:
We use `shared_ptr`, because it has a copy construct which is useful for merging projections. There we construct a bunch of MergeTasks inside a MergeTask with the same MinMaxIndex object. Before this there was a reference.